Unfixed strings, remove null termination

mfine opened the pull request:
Remove the fixed strings in fileio.  Remove the null termination description in settings.

/cc @fnoble @gsmcmullin @mookerji @denniszollo

swiftnav-jenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://zazu.ci.swiftnav.com/job/libsbp_Pull_Requests/202/
Test PASSed.

mfine commented:
This is missing the construct changes to support greedy strings. How do these C-structs looks WRT intended implementation?

mfine commented:
@fnoble what are your thoughts on removing the NULL-termination on settings and file_io messages? Right now, the code is checking everywhere for it and it will likely hurt a bit of backwards compat to drop it.

mfine commented:
@fnoble @gsmcmullin proposed new structure with a `sequence` field - comments, name field name suggestions welcome:

```diff
--- a/spec/yaml/swiftnav/sbp/file_io.yaml
+++ b/spec/yaml/swiftnav/sbp/file_io.yaml
@@ -37,6 +37,9 @@ definitions:
       invalid, a followup MSG_PRINT message will print "Invalid fileio
       read message".
     fields:
+      - sequence:
+          type: u32
+          desc: Sequence number for identifying request and response
       - offset:
           type: u32
           units: bytes
@@ -58,16 +61,12 @@ definitions:
       message where the message length field indicates how many bytes
       were succesfully read.
     fields:
-      - offset:
+      - sequence:
           type: u32
-          units: bytes
-          desc: File offset
-      - chunk_size:
-          type: u8
-          units: bytes
-          desc: Chunk size read
-      - filename_contents:
-          type: string
+          desc: Sequence number identifying request and response
+      - contents:
+          type: array
+          fill: u8
           desc: |
             Name of the file read from separated by NULL from its contents
```

mfine commented:
@fnoble @gsmcmullin as per discussion, the fileio messages have been updated to introduce a `sequence` number up front that gets echo'd back on responses. What do you guys think of these changes? No settings changes are needed!

fnoble commented on python/sbp/file_io.py:
```diff
@@ -47,28 +47,33 @@ class MsgFileioReadRequest(SBP):
                    ULInt32('offset'),
                    ULInt8('chunk_size'),
-                   String('filename', 20),)
+                   greedy_string('filename'),)
```
Does this get changed to `GreedyString` when your Construct PR makes it into upstream?

mfine replied:
Yeah, if/when it gets pulled in to upstream, we can merge pull `greedy_string`

fnoble commented on spec/yaml/swiftnav/sbp/file_io.yaml:
```diff
@@ -35,8 +35,12 @@ definitions:
       MSG_FILEIO_READ_RESPONSE message where the message length field
       indicates how many bytes were succesfully read. If the message is
       invalid, a followup MSG_PRINT message will print "Invalid fileio
```
Do you think we should remove the documentation around the `MSG_PRINT` response? Even if its true right now we don't want people to rely on it as part of the spec.

fnoble replied:
Ditto for the other instances

gsmcmullin replied:
I agree, it shouldn't be part of the spec.

mfine replied:
Definitely. Removing.

fnoble commented:
Like the sequence number! This looks good, seems to be missing updated C headers (and PDF, but I'm not sure if we have been generating that every time?)

swiftnav-jenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://zazu.ci.swiftnav.com/job/libsbp_Pull_Requests/207/
Test PASSed.

mfine commented:
Woops, missed the C generation. Added. Addressed other comments as well. Haven't been generating the doc every run. We'll regenerate around the release.

fnoble commented:
lgtm!

swiftnav-jenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://zazu.ci.swiftnav.com/job/libsbp_Pull_Requests/208/
Test PASSed.
